### DISCUSSION TITLE
Why does the shadcn-ui form embrace controlled inputs?

### DISCUSSION THREAD

**Discussion by joshuarost** [shadcn-ui/ui#3229] [Q&A]
React-hook-form has been written to embrace the use of uncontrolled components. Is there a reason why shadcn-ui uses the `Controller` component to make every field controlled? @shadcn

**Comment by KarimElsayad247** [comment]
I'm really wondering about the same thing.

It feels like a big philosophical clash, and I'd love to learn more about the rationale behind this decision. @shadcn

**Comment by govindpvenu** [comment]
Probably because of components like Select,Calendar which uses react-select,react-date-picker.
